minio/api.py

```diff
@@ -768,9 +771,13 @@ def put_object(self, bucket_name, object_name, data, length,
 
         metadata['Content-Type'] = 'application/octet-stream' if \
             not content_type else content_type
+        
+        if progress:
+            progress = Progress(total_size=length, file_name=object_name)
+
         if length > MIN_PART_SIZE:
             return self._stream_put_object(bucket_name, object_name,
-                                           data, length, metadata=metadata, sse=sse)
+                                           data, length, metadata=metadata, sse=sse, progress=progress)
```

Comment:
Overflowing the 80-100 column limit you should press enter and indent it within 80 column limit.